packages/components/src/components/hds/layout/flex/index.hbs

```diff
@@ -0,0 +1,11 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+{{#if (eq this.componentTag "div")}}
+  <div class={{this.classNames}} ...attributes>{{yield (hash Item=(component "hds/layout/flex/item"))}}</div>
+{{else}}
+  {{#let (element this.componentTag) as |Tag|}}<Tag class={{this.classNames}} ...attributes>{{yield
+        (hash Item=(component "hds/layout/flex/item"))
+      }}</Tag>{{/let}}
+{{/if}}
```

Comment:
@didoo Since the default tag is already a "div" (specified in the JS), does it need to be specified in the template here also using the "if/else" statements? That seems duplicative unless there's a reason I'm not aware of.

Reply:
I have noticed that when the the tag is generated dynamically, Ember adds a `ember-view` class to it, which is usually the one used for "components", while for the `<div>` default one it doesn't. 

I suspect that when it's dynamic, it's more javascript intense, is some kind of component, while in the other case it's just an HTML div. But I am making an educated guess, I'm not sure.

I have noticed this pattern here: https://github.com/prysmex/ember-eui/blob/a6ebc29d0c2b501472514d9dfff334a884a2e85b/packages/core/src/components/eui-flex-grid.gts#L43-L70

Reply:
@KristinLBradley let me know if you think is better to write a comment about it in the code (for future reference)

Reply:
Could be helpful to add a brief comment to explain

Reply:
@DingoEatingFuzz @aklkv before adding the comment in the code and write something which is not true, can you confirm my hunch that makes sense to add this `if` condition and why is that? thanks 🙏

Reply:
🙇‍♂️

Reply:
Added a comment about this.